curriculum/challenges/english/25-front-end-development/lecture-understanding-how-to-work-with-classes-in-javascript/673403ca2bb16658309e3632.md

```diff
@@ -10,6 +10,89 @@ dashedName: how-does-the-this-keyword-work
 
 Watch the lecture video and answer the questions below.
 
+# --transcript--
+
+How does the `this` keyword work?
+
+Let's learn about the `this` keyword. You learned that classes are like blueprints that you can use to create objects.
+
+These blueprints need a general way to refer to the specific object that is being created, accessed, or modified within the code. That's exactly what the `this` keyword is used for. It refers to the context where the code is supposed to run. If you use it within a method, the value of `this` will be a reference to the object associated to that method. This way, you can access the properties and methods of the object within the class and reuse the method on different objects.
+
+For example, here we have a `Dessert` class. This time, we're defining a `hasPeanuts` property.
+
+```js
+class Dessert {
+  constructor(hasPeanuts) {
+    this.hasPeanuts = hasPeanuts;
+  }
+}
+```
+
+Notice that we're using the `this` keyword within the constructor. Remember that the constructor is called automatically when a new object of the class is created.
+
+The `this` keyword is telling the program that `hasPeanuts` should be a property of the object that is currently being created.
+
+The right-hand side of the assignment statement is assigning the value of the `hasPeanuts` parameter to the `hasPeanuts` property of the new object.
+
+```js
+this.hasPeanuts = hasPeanuts;
+```
+
+That's in the context of the constructor but you can also use the `this` keyword within other methods.
+
+This is our `Dessert` class with a few changes. Now we have two properties: `name` and `price`.
+
+We also have a new `showPrice` method. In this method, we print a descriptive message to the console with the name and the price of the dessert object.
+
+Notice how we're using the `this` keyword to access the properties of the object within the `showPrice` method.
+
+```js
+class Dessert {
+  constructor(name, price) {
+    this.name = name;
+    this.price = price;
+  }
+
+  showPrice() {
+    console.log(`The price of ${this.name} is $${this.price}.`);
+  }
+}
+```
```

Comment:
````suggestion
This is our `Dessert` class with a few changes. Now we have two properties: `name` and `price`.

```js
class Dessert {
  constructor(name, price) {
    this.name = name;
    this.price = price;
  }

  showPrice() {
    console.log(`The price of ${this.name} is $${this.price}.`);
  }
}
```

We also have a new `showPrice` method. In this method, we print a descriptive message to the console with the name and the price of the dessert object.

Notice how we're using the `this` keyword to access the properties of the object within the `showPrice` method.
````